packages/plugin-ext/src/plugin/authentication-ext.ts

```diff
@@ -0,0 +1,207 @@
+
+    private _providerIds: string[] = [];
+
+    private _providers: theia.AuthenticationProviderInformation[] = [];
+
+    private onDidChangeAuthenticationProvidersEmitter = new Emitter<theia.AuthenticationProvidersChangeEvent>();
+    readonly onDidChangeAuthenticationProviders: Event<theia.AuthenticationProvidersChangeEvent> = this.onDidChangeAuthenticationProvidersEmitter.event;
+
+    private onDidChangeSessionsEmitter = new Emitter<theia.AuthenticationSessionsChangeEvent>();
+    readonly onDidChangeSessions: Event<theia.AuthenticationSessionsChangeEvent> = this.onDidChangeSessionsEmitter.event;
+
+    constructor(rpc: RPCProtocol) {
+        this.proxy = rpc.getProxy(PLUGIN_RPC_CONTEXT.AUTHENTICATION_MAIN);
+    }
+
+    getProviderIds(): Promise<ReadonlyArray<string>> {
+        return this.proxy.$getProviderIds();
+    }
+
+    get providerIds(): string[] {
+        return this._providerIds;
+    }
+
+    get providers(): ReadonlyArray<theia.AuthenticationProviderInformation> {
+        return Object.freeze(this._providers.slice());
+    }
+
+    async getSession(requestingExtension: InternalPlugin, providerId: string, scopes: string[],
+                     options: theia.AuthenticationGetSessionOptions & { createIfNone: true }): Promise<theia.AuthenticationSession>;
+    async getSession(requestingExtension: InternalPlugin, providerId: string, scopes: string[],
+                     options: theia.AuthenticationGetSessionOptions = {}): Promise<theia.AuthenticationSession | undefined> {
+        const provider = this.authenticationProviders.get(providerId);
+        const extensionName = requestingExtension.model.displayName || requestingExtension.model.name;
+        const extensionId = requestingExtension.model.id.toLowerCase();
+
+        if (!provider) {
+            throw new Error(`An authentication provider with id '${providerId}' was not found.`);
+        }
+
+        const orderedScopes = scopes.sort().join(' ');
+        const sessions = (await provider.getSessions()).filter(s => s.scopes.slice().sort().join(' ') === orderedScopes);
+
+        if (sessions.length > 0) {
+            if (!provider.supportsMultipleAccounts) {
+                const session = sessions[0];
+                const allowed = await this.proxy.$getSessionsPrompt(providerId, session.account.label, provider.label, extensionId, extensionName);
+                if (allowed) {
+                    return session;
+                } else {
+                    throw new Error('User did not consent to login.');
+                }
+            }
+
+            // On renderer side, confirm consent, ask user to choose between accounts if multiple sessions are valid
+            const selected = await this.proxy.$selectSession(providerId, provider.label, extensionId, extensionName, sessions, scopes, !!options.clearSessionPreference);
+            return sessions.find(session => session.id === selected.id);
```

Comment:
You're still sending the object you got from the plugin over the wire: this might not serialize correctly: you need to make a copy of the object to a well-known structure here.